ComponentKit/Core/Scope/CKComponentScopeHandle.mm

```diff
@@ -205,7 +207,6 @@ static Class controllerClassForComponentClass(Class componentClass)
     CKCAssert([controllerClass isSubclassOfClass:[CKComponentController class]],
               @"%@ must inherit from CKComponentController", controllerClass);
     CKComponentController *controller = [[controllerClass alloc] initWithComponent:component];
-
     [root registerAnnounceableEventsForController:controller];
```

Comment:
This is the bug, more or less. Since `CKComponentScopeHandle` only registers a component controller for receiving announceable events here the association will be lost when a new component scope root is created. Typically when a state update takes place.